tox.ini

```diff
@@ -28,7 +28,9 @@ commands =
     )\'
 # run tests
     python tests/manage.py test {env:TEST_SUITE:resolwe_bio} --noinput \
-        --verbosity 2 --parallel
+        --verbosity 2 --parallel \
+        --only-changes-to origin/{env:CHANGE_TARGET} \
```

Comment:
This won't work if a user runs Tox locally where the `CHANGE_TARGET` environment variable is not defined or on Jenkins when testing a branch that is not a pull request (e.g. `master`) where this environment variable is not defined.

Since Tox doesn't support conditional statements I would use its ability to create generative testing environments and define `-partial` testing environments:

```ini
envlist =
    py{27,34,35}{,-partial},
    docs,
    linters,
    packaging,
    extra,
    migrations
```

and then use this here to only conditionally include the added arguments:

```ini
# run tests
    python tests/manage.py test {env:TEST_SUITE:resolwe_bio} --noinput \
        # only test changes against a given Git commit reference for *-partial
        # testing environments (default to "master" branch)
        py{27,34,35}-partial: --only-changes-to {env:RESOLWE_TEST_ONLY_CHANGES_TO:master} \
        py{27,34,35}-partial: --changes-file-types .resolwebio-filetypes.yml \
        --verbosity 2 --parallel
```

The `Jenkinsfile` would need to be updated to handle this:
```
if (env.CHANGE_TARGET) {
    // run partial test suite depending on detected changes to the target
    // branch if we are testing a pull request
    withEnv(["RESOLWE_TEST_ONLY_CHANGES_TO=origin/${env.CHANGE_TARGET}"]) {
        sh "tox -e py34-partial"
    }
} else {
    sh "tox -e py34"
}
```

It would be nice if you also updated the documentation on [Using Tox](http://resolwe-bio.readthedocs.io/en/stable/contributing.html#using-tox) and mention the `-partial` version of test environments and how a developer can control the Git commit against which changes are calculated with `RESOLWE_TEST_ONLY_CHANGES_TO` environment variable.

Reply:
Good idea, done.